src/mapping/RadialBasisFctSolver.hpp

```diff
@@ -204,5 +216,88 @@ RadialBasisFctSolver::RadialBasisFctSolver(RADIAL_BASIS_FUNCTION_T basisFunction
     _qrMatrixQ = _matrixQ.colPivHouseholderQr();
   }
 }
+
+template <typename RADIAL_BASIS_FUNCTION_T>
+Eigen::VectorXd RadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>::solveConservative(const Eigen::VectorXd &inputData, Polynomial polynomial) const
+{
+  // TODO: Avoid temporary allocations
+  // Au is equal to the eta in our PETSc implementation
+  PRECICE_ASSERT(inputData.size() == _matrixA.rows());
+  Eigen::VectorXd Au = _matrixA.transpose() * inputData;
+  PRECICE_ASSERT(Au.size() == _matrixA.cols());
+
+  // mu in the PETSc implementation
+  Eigen::VectorXd out = _decMatrixC.solve(Au);
+
+  if (polynomial == Polynomial::SEPARATE) {
+    Eigen::VectorXd epsilon = _matrixV.transpose() * inputData;
+    PRECICE_ASSERT(epsilon.size() == _matrixV.cols());
+
+    // epsilon = Q^T * mu - epsilon (tau in the PETSc impl)
+    epsilon -= _matrixQ.transpose() * out;
+    PRECICE_ASSERT(epsilon.size() == _matrixQ.cols());
+
+    // out  = out - solveTranspose tau (sigma in the PETSc impl)
+#if EIGEN_VERSION_AT_LEAST(3, 4, 0)
```

Comment:
Can you comment on what feature was introduced in 3.4.0 that required so much extra work before?